Assets/Plugins/CountlySDK/Services/AbstractBaseService.cs

```diff
@@ -0,0 +1,22 @@
+﻿
+using System.Collections.Generic;
+using Plugins.CountlySDK.Enums;
+
+namespace Plugins.CountlySDK.Services
+{
+    public abstract class AbstractBaseService
+    {
+        internal List<AbstractBaseService> Listeners { get; set; }
+        protected readonly ConsentCountlyService _consentService;
+        
+
+        protected AbstractBaseService(ConsentCountlyService consentService)
+        {
+            _consentService = consentService;
+        }
+
+        internal virtual void DeviceIdChanged(string deviceId, bool merged) { }
+        internal virtual void ConsentChanged(Dictionary<Features, bool> updatedConsents) { }
```

Comment:
This function should have 2 parameters:
* an array of consents 
* a bool value that indicates the new consent value

Reply:
In Dictionary we have a new consent value against consent name. it's the same.
**Dictionary<Features, bool> updatedConsents**

Reply:
Even though both things are available in a dictionary, it is not the same. An array is a more compact data structure and it doesn't was memory on the same bool value for all changed consents.